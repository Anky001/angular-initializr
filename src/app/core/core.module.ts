import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { NoCacheHeadersInterceptor } from './interceptors/no-cache-header.interceptor';
import { HttpClientModule, HTTP_INTERCEPTORS } from '@angular/common/http';
import { AuthService } from './services/auth/auth.service';
import { StorageService } from './services/storage/storage.service';

@NgModule({
	declarations: [],
	imports: [CommonModule, HttpClientModule],
	providers: [
		{
			provide: HTTP_INTERCEPTORS,
			useClass: NoCacheHeadersInterceptor,
			multi: true,
		},
		AuthService,
		StorageService,
	],
})
export class CoreModule {}
