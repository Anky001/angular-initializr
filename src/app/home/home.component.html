<div class="container">
	<div fxLayout="column" fxLayoutAlign="center center">
		<form class="example-form">
			<mat-form-field class="example-full-width">
				<mat-label>User name</mat-label>
				<input matInput id="username" type="text" />
			</mat-form-field>
			<mat-form-field class="example-full-width">
				<mat-label>Password</mat-label>
				<input matInput id="password" type="password" />
			</mat-form-field>
			<button
				mat-raised-button
				color="warn"
				id="login-button"
				type="button"
				(click)="showLoginStatus = true"
			>
				Login
			</button>
		</form>
		<div class="login-status" *ngIf="showLoginStatus">Logged In</div>
	</div>
</div>
