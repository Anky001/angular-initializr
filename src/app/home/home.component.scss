.container {
  padding: 20px;
  background-color: var(--grey);
  h1 {
    margin: 0;
    color: #444;
  }
}
