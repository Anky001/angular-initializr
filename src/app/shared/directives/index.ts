// Import all the shared directives
export const directives: any[] = [];
