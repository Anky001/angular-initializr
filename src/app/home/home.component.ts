import { Component, OnInit } from '@angular/core';

@Component({
	selector: 'app-home',
	templateUrl: './home.component.html',
	styleUrls: ['./home.component.scss'],
})
export class HomeComponent implements OnInit {
	public showLoginStatus = false;
	constructor() {}

	ngOnInit(): void {
		this.firstToLaunch();
	}

	public firstToLaunch() {
		console.log('called');
	}
}
