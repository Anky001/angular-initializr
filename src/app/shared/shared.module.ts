import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { FlexLayoutModule } from '@angular/flex-layout';
import { RouterModule } from '@angular/router';
import { ReactiveFormsModule, FormsModule } from '@angular/forms';

import { MatInputModule } from '@angular/material/input';
import { MatDatepickerModule } from '@angular/material/datepicker';
import { MatTabsModule } from '@angular/material/tabs';
import { MatChipsModule } from '@angular/material/chips';
import { MatIconModule } from '@angular/material/icon';
import { MatToolbarModule } from '@angular/material/toolbar';
import { MatRippleModule } from '@angular/material/core';
import { MatButtonModule } from '@angular/material/button';
import { MatSnackBarModule } from '@angular/material/snack-bar';
import { MatCardModule } from '@angular/material/card';

import { HeaderComponent } from '@shared/components/header/header.component';

/**
 * Shared components
 */
export const components: any[] = [HeaderComponent];

/**
 * Angular Material modules
 */
const ngMaterialModules: any[] = [
  MatButtonModule,
  MatCardModule,
  MatChipsModule,
  MatDatepickerModule,
  MatIconModule,
  MatInputModule,
  MatRippleModule,
  MatTabsModule,
  MatToolbarModule,
  MatSnackBarModule,
];

@NgModule({
  declarations: [...components],
  imports: [
    CommonModule,
    FlexLayoutModule,
    RouterModule,
    FormsModule,
    ReactiveFormsModule,
    ...ngMaterialModules,
  ],
  exports: [
    FlexLayoutModule,
    FormsModule,
    ReactiveFormsModule,
    ...components,
    ...ngMaterialModules,
  ],
})
export class SharedModule {}
