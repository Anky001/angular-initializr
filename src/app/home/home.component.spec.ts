import { async, ComponentFixture, TestBed } from '@angular/core/testing';

import { HomeComponent } from './home.component';
import { SharedModule } from '@shared/shared.module';
import { TestingModule } from '@utils/testing.module';

describe('HomeComponent', () => {
	let component: HomeComponent;
	let fixture: ComponentFixture<HomeComponent>;

	beforeEach(async(() => {
		TestBed.configureTestingModule({
			imports: [SharedModule, TestingModule],
			declarations: [HomeComponent],
		}).compileComponents();
	}));

	beforeEach(() => {
		fixture = TestBed.createComponent(HomeComponent);
		component = fixture.componentInstance;
		fixture.detectChanges();
	});

	it('should create', () => {
		expect(component).toBeTruthy();
	});

	it('should check if showLoginStatus is false', () => {
		expect(component.showLoginStatus).toBeFalsy();
	});

	it('should call firstToLaunch() after ngOnInit is called', () => {
		const spy = jest.spyOn(component, 'firstToLaunch');
		component.ngOnInit();
		expect(spy).toHaveBeenCalled();
	});

	it('should match the snapshot', () => {
		expect(component).toMatchSnapshot();
	});
});
