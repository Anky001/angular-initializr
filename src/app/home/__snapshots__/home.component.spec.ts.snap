// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeComponent should match the snapshot 1`] = `
HomeComponent {
  "showLoginStatus": false,
}
`;
